bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -821,6 +821,9 @@ public void scanJournal(long journalId, long journalPos, JournalScanner scanner)
                     scanner.process(journalVersion, offset, recBuff);
                 }
             }
+            // Update LastLogMark to EOF position after replaying journal
+            // After force flush of LedgerStorage, SyncThread will persist this to disk
+            lastLogMark.setCurLogMark(journalId, recLog.fc.position());
```

Comment:
@karanmehta93 in retrospect, i think this may not be right place to update the lastLogMark. It should be part of the scanner (JournalScanner) logic. Not all scanners add entries to LedgerStorage, so this update logic should be part of JournalScanner, which is created in Bookie.readJournal method.

Reply:
Yes @reddycharan That's a good observation. As discussed offline, I am refactoring the code to move the `replay()` method from Journal to Bookie, since it's bookie's task to replay the journal. In that manner we can simplify the logic of updating `LastLogMark`. 
By doing this, we lose the information of the file pointer end, hence I am arbitrarily updating it to `LONG.MAX_VALUE` since the FileChannel treats both those cases similarly.